modules/apps/blogs/blogs-web/src/main/resources/META-INF/resources/blogs/js/blogs.js

```diff
@@ -12,769 +12,616 @@
-						savedAtMessage: Liferay.Language.get(
-							'entry-saved-at-x'
-						),
-						savedDraftAtMessage: Liferay.Language.get(
-							'draft-saved-at-x'
-						),
-						titleRequiredAtPublish: Liferay.Language.get(
-							'this-field-is-required-to-publish-the-entry'
-						),
-					},
-				},
-			},
-
-			AUGMENTS: [Liferay.PortletBase],
-
-			EXTENDS: A.Base,
-
-			NAME: 'liferay-blogs',
-
-			NS: 'liferay-blogs',
-
-			prototype: {
-				_automaticURL() {
-					return (
-						this.one('#urlOptions').one('input:checked').val() ===
-						'true'
-					);
-				},
-
-				_beforePublishBtnClick() {
-					var instance = this;
-
-					var form = Liferay.Form.get(instance.ns('fm'));
-
-					var strings = instance.get('strings');
-
-					form.addRule(
-						instance.one('#title'),
-						'required',
-						strings.titleRequiredAtPublish
-					);
-				},
-
-				_beforeSaveBtnClick() {
-					var instance = this;
-
-					var form = Liferay.Form.get(instance.ns('fm'));
-
-					form.removeRule(instance.one('#title'), 'required');
-				},
-
-				_bindUI() {
-					var instance = this;
-
-					instance._captionNode = instance.one(
-						'.cover-image-caption'
-					);
-
-					var eventHandles = [
-						Liferay.on(
-							'coverImageDeleted',
-							instance._removeCaption,
-							instance
-						),
-						Liferay.on(
-							['coverImageUploaded', 'coverImageSelected'],
-							instance._showCaption,
-							instance
-						),
-					];
-
-					var publishButton = instance.one('#publishButton');
-
-					if (publishButton) {
-						eventHandles.push(
-							publishButton.before(
-								STR_CLICK,
-								A.bind('_beforePublishBtnClick', instance)
-							),
-							publishButton.on(
-								STR_CLICK,
-								A.bind(
-									'_checkImagesBeforeSave',
-									instance,
-									false,
-									false
-								)
-							)
-						);
-					}
-
-					var saveButton = instance.one('#saveButton');
-
-					if (saveButton) {
-						eventHandles.push(
-							saveButton.before(
-								STR_CLICK,
-								A.bind('_beforeSaveBtnClick', instance)
-							),
-							saveButton.on(
-								STR_CLICK,
-								A.bind(
-									'_checkImagesBeforeSave',
-									instance,
-									true,
-									false
-								)
-							)
-						);
-					}
+import EventListener from './utils/EventListener';
+
+const CSS_INVISIBLE = 'invisible';
+const STR_BLANK = '';
+const STR_CHANGE = 'change';
+const STR_CLICK = 'click';
+const STR_SUFFIX = '...';
+
+const STRINGS = {
+	confirmDiscardImages: Liferay.Language.get(
+		'uploads-are-in-progress-confirmation'
+	),
+	saveDraftError: Liferay.Language.get('could-not-save-draft-to-the-server'),
+	saveDraftMessage: Liferay.Language.get('saving-draft'),
+	savedAtMessage: Liferay.Language.get('entry-saved-at-x'),
+	savedDraftAtMessage: Liferay.Language.get('draft-saved-at-x'),
+	titleRequiredAtPublish: Liferay.Language.get(
+		'this-field-is-required-to-publish-the-entry'
+	),
+};
+
+function nsObj(obj, namespace) {
```

Comment:
In general, try to [avoid abbreviations](https://github.com/liferay/liferay-frontend-guidelines/blob/master/general/naming.md#avoid-abbreviations)